src/components/playground/Playground/Playground.js

```diff
@@ -110,6 +110,22 @@ class Playground extends Component {
   onSubmitFlagsFormSwitch = () =>
     this.setState({ isEditingFlags: !this.state.isEditingFlags })
 
+  onResetConditionsButtonClick = () => {
+    this.setState({
+      chunks: [],
+      isEditingAtIndex: null,
+      isEditingFlags: false,
+      flags: {},
+      lastActiveSentenceMenu: null,
+      regexChunks: [],
+      sourceString: "",
+      sentenceEl: null,
+    })
+    while (this.core.getChunks().length !== 0) {
+      this.core.deleteCondition(0)
+    }
```

Comment:
This is nice @drwm-base! 👍 
It removes all the conditions as expected!

However, it resets the flags too (line 118). Am I right? Is this may delete an enabled _“global flag”_?

Also, have a noticed that the core manages the adds, updates and single deletions? Maybe I should write this in guidelines, my bad. So, in your new code, the loop is in _Playground.js_, which makes the component _Playground_ responsible of its own _this.core_'s conditions array. What do you think about moving this responsibility to the _core.js_ file by creating in it a new method (_removeAllConditions_ for instance) and calling this new core instance method from _Playground.onResetConditionsButtonClick_?

Reply:
I thought that would be a good idea when I was writing, but I wasn't sure as to the scope that you wanted me to cover (ie just add a new button / changes to the core or just playground.js). I can move the functionality over to the core.js file, it would make things look cleaner as well. I will do that asap!

Thanks for the feedback!